drv/lpc55-update-server/src/main.rs

```diff
@@ -1027,6 +1100,71 @@ impl ServerImpl<'_> {
 
         Ok(())
     }
+
+    /// Cancel pending transient or persistent Hubris image selection.
+    fn switch_default_hubris_image_cancel_pending(
+        &mut self,
+        slot: SlotId,
+        duration: SwitchDuration,
+    ) -> Result<(), RequestError<UpdateError>> {
+        // Note: Rollback policy will be checking epoch values before activating.
+        match duration {
+            SwitchDuration::Once => {
+                // Is there a pending transient boot selection?
+                // If it matches the one being cancelled
+                if let Some(transient_selection) =
+                    get_hubris_transient_override()
+                {
+                    if transient_selection == slot {
+                        set_hubris_transient_override(None);
+                        Ok(())
+                    } else {
+                        Err(UpdateError::InvalidPreferredSlotId.into())
+                    }
+                } else {
+                    Err(UpdateError::NonePending.into())
+                }
+            }
```

Comment:
I'm not a huge fan of requiring the exact slot for cancellation. I could see an argument that it's more robust but at the same time it's just as easy to make a mistake passing in an incorrect slot. I'd argue that it's safer to require fewer options vs having to handle your cancellation failing. I'd really prefer to just pass the component and the persistent option but I'm not going to block the PR over it.